ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufDataTranslator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+
+package io.confluent.ksql.serde.protobuf;
+
+import io.confluent.ksql.serde.SchemaFullNameAppender;
+import io.confluent.ksql.serde.connect.ConnectDataTranslator;
+import io.confluent.ksql.serde.connect.DataTranslator;
+import io.confluent.ksql.util.DecimalUtil;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+
+public class ProtobufDataTranslator implements DataTranslator {
+  private final DataTranslator innerTranslator;
+  private final Schema ksqlSchema;
+  private final Schema protoCompatibleSchema;
+
+  ProtobufDataTranslator(final Schema schema, final String schemaFullName) {
+    this.ksqlSchema = Objects.requireNonNull(schema, "schema");
+
+    this.protoCompatibleSchema = SchemaFullNameAppender.appendSchemaFullName(
+        schema, schemaFullName
+    );
+
+    this.innerTranslator = new ConnectDataTranslator(protoCompatibleSchema);
+  }
+
+  Schema getProtoCompatibleSchema() {
+    return protoCompatibleSchema;
+  }
+
+  @Override
+  public Object toKsqlRow(final Schema connectSchema, final Object connectObject) {
+    final Object protoCompatibleRow = innerTranslator.toKsqlRow(connectSchema, connectObject);
+    if (protoCompatibleRow == null) {
+      return null;
+    }
+
+    return replaceSchema(ksqlSchema, protoCompatibleRow);
```

Comment:
These were for avro schema. 

> Responsible for converting the KSQL schema to a version ready for connect to convert to an
avro schema.
This includes ensuring field names are valid Avro field names and that nested types do not
have name clashes.

Do we need this for protobuf? I'm concerned we are changing too much which is not necessary.

Reply:
Replacing the schema to add the full schema name is the main thing needed to make the INSERT to work. The Connect serializer checks the row schema and try to match it with SR. If it has a different name, then it fails.

I debugged this when the `toConnectRow` is called, though. I haven't found a test that calls the `toKsqlRow` to verify it. Any ideas?